content/influxdb/cloud-serverless/reference/client-libraries/v3/python.md

```diff
@@ -8,17 +8,490 @@ menu:
+
+### Syntax
+
+```py
+__init__(self, host=None, org=None, database=None, token=None,
+        _write_client_options=None, _flight_client_options=None, **kwargs)
+```
+
+Initializes and returns an `InfluxDBClient3` instance with the following:
+
+- A singleton _write client_ configured for writing to the database.
+- A singleton _Flight client_ configured for querying the database.
+
+### Attributes
+
+- **org** (str): The organization name (for {{% cloud-name %}}, set this to an empty string (`""`)).
+- **database** (str): The database to use for writing and querying.
+- **_write_client_options** (dict): Options to use when writing to InfluxDB.
+  If `None`, writes are [synchronous](#synchronous-writing).
+- **_flight_client_options** (dict): Options to use when querying InfluxDB.
+
+#### Batch writing
+
+In batching mode, the client adds the record or records to a batch, and then schedules the batch for writing to InfluxDB.
+The client writes the batch to InfluxDB after reaching `_write_client_options.batch_size` or `_write_client_options.flush_interval`.
+If a write fails, the client reschedules the write according to the `_write_client_options` retry options.
+
+To use batching mode, pass an instance of `WriteOptions` for the `InfluxDBClient3._write_client_options` argument--for example:
+
+1.  Instantiate `WriteOptions()` with defaults or with
+`WriteOptions.write_type=WriteType.batching`.
+
+    ```py
+      # Batching with all batch size, flush, and retry defaults
+      write_options = WriteOptions()
+    ```
+
+2.  Call the a `write_client_options` function to create an options object that uses `write_options` from the preceding step.
+
+    ```py
+    wco = write_client_options(WriteOptions=write_options)
+    ```
+
+3.  Initialize the client, setting the `_write_client_options` argument to the `wco` object from the preceding step.
+
+    {{< tabs-wrapper >}}
+{{% code-placeholders "BUCKET_(NAME|TOKEN)|API_TOKEN" %}}
+```py
+with  InfluxDBClient3(token="API_TOKEN", host="cloud2.influxdata.com",
+                      org="", database="BUCKET_NAME",
+                      _write_client_options=wco) as client:
+
+      client.write(record=points)
+```
+{{% /code-placeholders %}}
+    {{< /tabs-wrapper >}}
+
+#### Synchronous writing
+
+In synchronous mode, the client sends write requests immediately (not batched)
+and doesn't retry failed writes.
+
+To use synchronous mode, set `_write_client_options=None` or `_write_client_options.write_type=WriteType.synchronous`.
+
+### Examples
+
+##### Initialize a client
+
+The following example initializes a client for writing and querying the bucket.
+Given `_write_client_options=None`, the client will use synchronous mode when writing data.
+
+{{% code-placeholders "BUCKET_(NAME|TOKEN)|API_TOKEN" %}}
+```py
+client = InfluxDBClient3(token="API_TOKEN",
+                         host="cloud2.influxdata.com",
+                         org="",
```

Comment:
Do they need their org name with serverless or is it inferred from the token?

Reply:
> Do they need their org name with serverless or is it inferred from the token?

Inferred from the token. Just tested again and it works when `org` is omitted.

```py
>>> client = InfluxDBClient3(host=host, org="", token=token, database=database)
>>> client.query('show tables')
pyarrow.Table
table_catalog: string not null
table_schema: string not null
table_name: string not null
table_type: string not null
----
table_catalog: [["public","public","public","public","public",...,"public","public","public","public","public"]]
table_schema: [["iox","iox","iox","iox","iox",...,"iox","information_schema","information_schema","information_schema","information_schema"]]
table_name: [["earthquake","explosion","home","homeSensor","ice quake",...,"temperature","tables","views","columns","df_settings"]]
table_type: [["BASE TABLE","BASE TABLE","BASE TABLE","BASE TABLE","BASE TABLE",...,"BASE TABLE","VIEW","VIEW","VIEW","VIEW"]]
>>> client = InfluxDBClient3(host=host, token=token, database=database)
>>> client.query('show tables')
pyarrow.Table
table_catalog: string not null
table_schema: string not null
table_name: string not null
table_type: string not null
----
table_catalog: [["public","public","public","public","public",...,"public","public","public","public","public"]]
table_schema: [["iox","iox","iox","iox","iox",...,"iox","information_schema","information_schema","information_schema","information_schema"]]
table_name: [["earthquake","explosion","home","homeSensor","ice quake",...,"temperature","tables","views","columns","df_settings"]]
table_type: [["BASE TABLE","BASE TABLE","BASE TABLE","BASE TABLE","BASE TABLE",...,"BASE TABLE","VIEW","VIEW","VIEW","VIEW"]]
>>> 
```